packages/ui-components/src/components/Switch/Switch.component.tsx

```diff
@@ -0,0 +1,310 @@
+   */
+  disabled?: boolean
+  /**
+   * Indicates if the Switch has an error state
+   */
+  invalid?: boolean
+  /**
+   * Indicates if the Switch has been validated successfully
+   */
+  valid?: boolean
+  /**
+   * Help text to provide additional information about the Switch
+   */
+  helptext?: React.ReactNode
+  /**
+   * Text to display when the Switch has validation errors
+   */
+  errortext?: React.ReactNode
+  /**
+   * Text to display when the Switch has been validated successfully
+   */
+  successtext?: React.ReactNode
+  /**
+   * Custom class name to apply to the internal button element
+   */
+  className?: string
+  /**
+   * Callback function to handle changes to the switch state
+   */
+  // eslint-disable-next-line no-unused-vars
+  onChange?: (event: React.ChangeEvent<HTMLButtonElement>) => void
+  /**
+   * Callback function to handle click events on the switch
+   */
+  // eslint-disable-next-line no-unused-vars
+  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void
+  /**
+   * Custom class name to apply to the wrapper element
+   * Useful for styling and positioning the outermost element of the component
+   */
+  wrapperClassName?: string
+}
+
+const isNotEmptyString = (str: unknown): str is string => {
+  return typeof str === "string" && str.trim().length > 0
+}
+
+const getSwitchSizeStyles = (size: SwitchSize): string => {
+  switch (size) {
+    case "small":
+      return "jn-w-[1.75rem] jn-h-4"
+    case "large":
+      return "jn-w-[3.125rem] jn-h-[1.6875rem]"
+    default:
+      return "jn-w-switch-default jn-h-switch-default"
+  }
+}
+
+const getHandleSizeStyles = (size: SwitchSize): string => {
+  switch (size) {
+    case "small":
+      return "jn-w-[0.75rem] jn-h-[0.75rem]"
+    case "large":
+      return "jn-w-[1.4375rem] jn-h-[1.4375rem]"
+    default:
+      return "jn-w-switch-handle-default jn-h-switch-handle-default"
+  }
+}
+
+const renderValidationIcon = (isInvalid: boolean, isValid: boolean, disabled: boolean): React.ReactNode | null => {
+  if (!isInvalid && !isValid) return null
+
+  if (isValid)
+    return (
+      <Icon
+        icon="checkCircle"
+        color="jn-text-theme-success"
+        size="1.125rem"
+        className={`${iconBaseStyles} ${disabled ? "jn-opacity-50" : ""}`}
+      />
+    )
+
+  return (
+    <Icon
+      icon="dangerous"
+      color="jn-text-theme-error"
+      size="1.125rem"
+      className={`${iconBaseStyles} ${disabled ? "jn-opacity-50" : ""}`}
+    />
+  )
+}
+
+const renderFormHint = (
+  errortext: React.ReactNode,
+  successtext: React.ReactNode,
+  helptext: React.ReactNode
+): React.ReactNode | null => {
+  if (isNotEmptyString(errortext)) return <FormHint text={errortext} variant="error" className={hintBaseStyles} />
+
+  if (isNotEmptyString(successtext)) return <FormHint text={successtext} variant="success" className={hintBaseStyles} />
+
+  if (isNotEmptyString(helptext)) return <FormHint text={helptext} className={hintBaseStyles} />
```

Comment:
if this thing had variant we could refactoring this but not now I think  it is a good one👍

Reply:
That's a good idea. Will take note.